src/Tracing/SentryConsoleTransactionAdapter.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry\Tracing;
+
+use Psr\Log\LoggerInterface;
+use Sentry\SentrySdk;
+use Sentry\Tracing\SpanContext;
+use Sentry\Tracing\TransactionContext;
+use Yiisoft\Yii\Console\Event\ApplicationShutdown;
+
+final class SentryConsoleTransactionAdapter
+{
+    public function __construct(private LoggerInterface $logger, private SentryTraceConsoleListener $consoleListener)
+    {
+    }
+
+    public function begin(?string $sentryTraceString = null): self
+    {
+        $hub = SentrySdk::getCurrentHub();
+        if ($sentryTraceString) {
+            $context = TransactionContext::fromSentryTrace($sentryTraceString);
+        } else {
+            $context = new TransactionContext();
+        }
+        $context->setOp('console sub task');
+
+        $context->setStartTimestamp(microtime(true));
+
+        $transaction = $hub->startTransaction($context);
+        $transaction->setName('undefined command');
```

Comment:
Why it's undefined here?

Reply:
user must set name properly for setName(). Useing this class is optional.